cpp/benchmarks/t/pipelines/registration/Registration.cpp

```diff
@@ -0,0 +1,183 @@
+
+// Initial transformation guess for registation.
+static const std::vector<float> initial_transform_flat{
+        0.862, 0.011, -0.507, 0.5,  -0.139, 0.967, -0.215, 0.7,
+        0.487, 0.255, 0.835,  -1.4, 0.0,    0.0,   0.0,    1.0};
+
+namespace open3d {
+namespace t {
+namespace pipelines {
+namespace registration {
+
+static std::tuple<geometry::PointCloud, geometry::PointCloud>
+LoadTensorPointCloudFromFile(const std::string& source_pointcloud_filename,
+                             const std::string& target_pointcloud_filename,
+                             const double voxel_downsample_factor,
+                             const core::Dtype& dtype,
+                             const core::Device& device) {
+    geometry::PointCloud source, target;
+
+    io::ReadPointCloud(source_pointcloud_filename, source,
+                       {"auto", false, false, true});
+    io::ReadPointCloud(target_pointcloud_filename, target,
+                       {"auto", false, false, true});
+
+    // Eliminates the case of impractical values (including negative).
+    if (voxel_downsample_factor > 0.001) {
+        // TODO: Use geometry::PointCloud::VoxelDownSample.
+        open3d::geometry::PointCloud legacy_s = source.ToLegacyPointCloud();
+        open3d::geometry::PointCloud legacy_t = target.ToLegacyPointCloud();
+
+        legacy_s = *legacy_s.VoxelDownSample(voxel_downsample_factor);
+        legacy_t = *legacy_t.VoxelDownSample(voxel_downsample_factor);
+
+        source = geometry::PointCloud::FromLegacyPointCloud(legacy_s);
+        target = geometry::PointCloud::FromLegacyPointCloud(legacy_t);
+    } else {
+        utility::LogWarning(
+                " VoxelDownsample: Impractical voxel size [< 0.001], skiping "
+                "downsampling.");
+    }
+
+    geometry::PointCloud source_device(device), target_device(device);
+
+    core::Tensor source_points = source.GetPoints().To(device, dtype);
+    source_device.SetPoints(source_points);
+
+    core::Tensor target_points = target.GetPoints().To(device, dtype);
+    core::Tensor target_normals = target.GetPointNormals().To(device, dtype);
+    target_device.SetPoints(target_points);
+    target_device.SetPointNormals(target_normals);
+
+    return std::make_tuple(source_device, target_device);
+}
+
+static void BenchmarkRegistrationICP(benchmark::State& state,
+                                     const core::Device& device,
+                                     const TransformationEstimationType& type) {
+    core::Dtype dtype = core::Dtype::Float32;
+
+    geometry::PointCloud source(device), target(device);
+
+    std::tie(source, target) = LoadTensorPointCloudFromFile(
+            source_pointcloud_filename, target_pointcloud_filename,
+            voxel_downsampling_factor, dtype, device);
+
+    core::Tensor init_trans =
+            core::Tensor(initial_transform_flat, {4, 4}, dtype, device);
+
+    RegistrationResult reg_result(init_trans);
```

Comment:
move declaration to right before its first use

Reply:
Done.